src/librustc_driver/lib.rs

```diff
@@ -593,6 +594,19 @@ fn show_content_with_pager(content: &String) {
 }
 
 impl RustcDefaultCalls {
+    pub fn process_rlink(sess: &Session, compiler: &interface::Compiler) -> Compilation {
```

Comment:
This isn't feature gated. With a hand made json file it is possible to use this.

Reply:
Added new flag `-Z link-only`